src/get-started/java-binding.md

```diff
@@ -491,75 +493,201 @@ as specified in [How to Run a Service section](#how-to-run-a-service).
-    import org.junit.Before;
-    import org.junit.Test;
+To instantiate a TestKit use [`TestKit.Builder`][testkit-builder]. It allows
+configuration of:
 
-    public class MySchemaTest {
+- Type of emulated node (either [Validator][validator] or [Auditor][auditor])
+- Services with which the TestKit would be instantiated
+- [`TimeProvider`][testkit-time-provider] if usage of [Time Oracle](../advanced/time.md)
+is needed (for details see [Time Oracle Testing](#time-oracle-testing))
+- Number of validators in emulated network
 
-      MemoryDb db;
-      Cleaner cleaner;
+!!! note
+    Note that regardless of the configured number of validators, only a single
+    node will be emulated. This node will create the service instances, execute
+    their operations (e.g.,
+    [`afterCommit(BlockCommittedEvent)`][service-after-commit] method logic),
+    and provide access to its state.
 
-      @Before
-      public void setUp() {
-        db = MemoryDb.newInstance();
-        cleaner = new Cleaner();
-      }
+Default TestKit can be instantiated with a single validator as an emulated node,
+single service and without Time Oracle in the following way:
 
-      @After
-      public void tearDown() throws Exception {
-        cleaner.close();
-        db.close();
-      }
+```java
+try (TestKit testKit = TestKit.forService(MyServiceModule.class)) {
+  // Test logic
+}
+```
 
-      @Test
-      public void testCreatingSnapshot() {
-        Snapshot snapshot = db.createSnapshot(cleaner);
-        // Use the Snapshot to test the schema.
-      }
+Or using a builder if different configuration is needed:
 
-      @Test
-      public void testMergingSomeChanges() {
-        Fork fork = db.createFork(cleaner);
-        // Make some changes to Fork.
-        // …
-
-        // Merge the changes made to the database Fork into the database.
-        db.merge(fork);
-
-        // Create a Snapshot of the database state. Must include the changes
-        // made above.
-        Snapshot snapshot = db.createSnapshot(cleaner);
-        // Use the Snapshot to test the schema/transaction
-        // with a certain database state.
-        // …
-      }
-    }
-    ```
+```java
+try (TestKit testKit = TestKit.builder()
+    .withServices(MyServiceModule.class, MyServiceModule2.class)
+    .withValidators(2)
+    .build()) {
+  // Test logic
+}
+```
 
-### Transactions
+### Transactions testing
 
-To test transactions execution, you can use `MemoryDb`, as in the previous
-section.
+TestKit allows testing transaction execution by submitting blocks with given
+[transaction messages][transactions-messages]:
+
+```java
+try (TestKit testKit = TestKit.forService(MyServiceModule.class)) {
+  // Construct some transaction to be executed
+  TransactionMessage message = constructTransactionMessage();
+  // Commit block with this transaction
+  Block block = testKit.createBlockWithTransactions(message);
+  // Check the resulting block or blockchain state
+}
+```
+
```

Comment:
I'd consider adding an example of testing transactions throwing TransactionExecutionException because it is not apparent.

Reply:
Wouldn't it be clearer if that was presented in two separate snippets of code? These are the first examples of testkit usage, hence I'd strive for clarity.

Also, in the first example with a valid transaction, I'd add a comment explaining that the snapshot can also be used to verify the state changes made by this transaction (as that's what usual tests do). Or even leave out the assertion on the execution result, what do you think? Or keep it, but with more detailed comments:
```java
// A test can obtain a snapshot of the current database state:
Snapshot snapshot = testkit.getSnapshot();
// It can be used to access the core schema, e.g., to check the transaction 
// execution result:
Optional<TransactionResult> validTxResult =
      blockchain.getTxResult(validTx.hash());
assertThat(validTxResult).hasValue(TransactionResult.successful());
// ... and also to verify the changes the transaction made to the service state:
MySchema schema = new MySchema(snapshot);
// Perform assertions on the data in the service schema.
```

Reply:
I actually had those as separate snippets at first, but then merged as they had mostly duplicated code. Will change that